public/js/post/post.js

```diff
@@ -4,11 +4,28 @@ var CourseCode = React.createClass({
     },
 
     componentWillMount: function() {
-        this.setState({selected: getCookie(this.props.courseID) === 'active'});
+        this.setState({selected: getCookie(this.props.courseIDs[0]) === 'active'});
     },
 
     toggleFullInfo: function() {
-        $('#' + this.props.courseID + '_info').toggle();
+        $('#' + this.props.courseIDs[0] + '_info').toggle();
+    },
+
+    getCategoryName: function() {
+        var categoryName = '';
+
+        if (this.props.courseIDs[0] === 'mat135') {
+            // special case for calculus requirement since it doesn't fit the same pattern
+            categoryName = '(MAT135H and MAT136H) or MAT137Y or MAT157Y';
+        } else {
+            categoryName += this.props.courseIDs[0].toUpperCase() + 'H';
+
+            for (i = 1; i < this.props.courseIDs.length; i++) {
+                categoryName += " or " + this.props.courseIDs[i].toUpperCase() + 'H';
```

Comment:
1) Maybe start by doing a map onto `this.props.courseIDs` to convert to uppercase and add an 'H'.
2) Use the Javascript `join` method for arrays.